### DISCUSSION TITLE
Control over missing key error handling

### DISCUSSION THREAD

**Discussion by rpspall** [i18next/i18next#2088] [Q&A]
I won't bore with the details, but I need to be able to control the logic for "missing keys" in i18next react. All of the keys in my application currently end with **.pv** (there is a good explanation for this, but ultimately doesn't add to the conversation!), e.g. 'mypage.mybutton.pv', 'mypage.mytitle.pv'

With a migration that it happening, this .pv is being removed from 80 percent of the keys, but still remains for the other 20 percent - slowly being removed in the remaining 20 percent as well! Rather than iterating through the thousands of i18n keys and figuring out which ones have had he .pv removed and which haven't I want to globally remove ALL the **.pv** suffixes from our UI and then implement some error handling which basically says...

When looking up the translation for a key if the key isn't found, fall back to looking up the key with a **.pv** suffix...

Basically if my code looks up a translation for key 'myScreen.title' and doesn't find the key, I want it to fallback and lookup 'myScreen.title.pv'. Obviously this is temporary logic, and we will eventually remove all **.pv** suffixes from our i18n key file and then the logic can be removed.

Any suggestions?

Current React init code:

```
//There is no isInitialized flag / logic, so to avoid double initialization, check
    //If one of the i18next options has been set
    const languageNames = application.avaliableLanguages.map((avaliableLanguage) => avaliableLanguage.language);
    if (!i18next.options.fallbackLng) {
        i18next
            // detect user language
            // learn more: https://github.com/i18next/i18next-browser-languageDetector
            .use(I18nextBrowserLanguageDetector)
            // pass the i18n instance to react-i18next.
            .use(initReactI18next)
            .use(I18NextHttpBackend)
            // init i18next
            // for all options read: https://www.i18next.com/overview/configuration-options
            .init({
                debug: false,
                returnNull: false,
                backend: {
                    loadPath: determineI18NLoadPath(application.applicationRoot),
                },
                fallbackLng: application.fallbackLanguage,
                //Ultimately, the backend decides which locales are actually supported on a web module basis!
                supportedLngs: languageNames,
                fallbackNS: application.applicationRoot,
                ns: application.applicationRoot,
                interpolation: {
                    escapeValue: false, // not needed for react as it escapes by default
                    prefix: '{',
                    suffix: '}',
                },
                react: {
                    transKeepBasicHtmlNodesFor: ['br', 'strong', 'i', 'b', 'p', 'span', 'ol', 'li'],
                },
            });
    }
```

Current react i18n translation code:

```
import { useTranslation } from 'react-i18next';
...
 const { t: translate } = useTranslation();
....
<div style={{ fontWeight: 600 }}>{translate('page.storeHours.openingTime')}</div>
...

```

**Comment by jamuhl** [comment]
could be done using a postProcessor -> eg. https://github.com/i18next/i18next-intervalPlural-postProcessor/blob/master/src/index.js#L62

**Comment by rpspall** [comment]
Sounds promising... So I can implement my own post processor... will give it a try... thank you
